CDS-95041-CD Release Notes version 1.32.x (#6200)
harness/developer-hub: release-notes/continuous-delivery.md

## April 2024

### Version 1.32.x

#### Fixed issues

- Fixed an issue where the Nexus 2 artifactory registry drop-down listed duplicate group IDs. (CDS-94376, ZD-60041)

> - Terraform deployment failed when using AWS connectors (IRSA credential type with assume cross account role) in Terraform steps. When the Terraform Apply command duration exceeded 15 minutes, Terraform output threw an error saying the credentials are invalid when trying to assume a different role from the AWS backend configuration. The default duration for assuming the role in the `aws-java-sdk` is 15 minutes. But the SDK allowed to override it causing the issue. This issue is resolved by introducing a new Harness variable, `HARNESS_AWS_ASSUME_ROLE_DURATION`. When you set this environment variable in Terraform steps, the value mentioned in the variable overrides the default 15 minutes value to assume AWS roles. This item requires Harness Delegate version 01.04.827xx. For information about features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate).(CDS-94355, ZD-60095)

Reviewer: This seemed really hard to follow somehow. Any way to make it clearer please?

> - Terraform deployment failed when using AWS connectors (IRSA credential type with assume cross account role) in Terraform steps. When the Terraform Apply command duration exceeded 15 minutes, Terraform output threw an error saying the credentials are invalid when trying to assume a different role from the AWS backend configuration. The default duration for assuming the role in the `aws-java-sdk` is 15 minutes. But the SDK allowed to override it causing the issue. This issue is resolved by introducing a new Harness variable, `HARNESS_AWS_ASSUME_ROLE_DURATION`. When you set this environment variable in Terraform steps, the value mentioned in the variable overrides the default 15 minutes value to assume AWS roles. This item requires Harness Delegate version 01.04.827xx. For information about features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate).(CDS-94355, ZD-60095)

Author: ```suggestion
- Terraform deployment failed when using AWS connectors (IRSA credential type with assume cross account role) in Terraform steps. This issue occurred when the Terraform Apply was trying to assume a different role from the AWS backend configuration. The default duration for assuming the role in the `aws-java-sdk` is 15 minutes. When the Terraform Apply step exceeded 15 minutes, the Terraform output threw an error. This issue is resolved by introducing a new Harness variable, `HARNESS_AWS_ASSUME_ROLE_DURATION`. In Terraform steps, you can now set the environment variable value to override the default duration to assume AWS roles. This item requires Harness Delegate version 01.04.827xx. For information about features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate).(CDS-94355, ZD-60095)
```

> - The RepoName, FilePath, and ConnectorRef parameters are marked as required in the Git import APIs for pipelines, templates, input sets, services, environments, infrastructure definitions, and service overrides. These parameters were optional before but made mandatory now as the APIs require these to work. (CDS-94245)

Reviewer: This one is a breaking change. Do we want to add any warnings/extra text around it to make it prominent?

Author: Should this be a new enhancement?

Reviewer: Not an enhancement, but a breaking change still. Customers are getting nothing new from this.

## April 2024

### Version 1.33.3

#### Breaking changes

- The RepoName, FilePath, and ConnectorRef parameters are marked as required in the Git import APIs for pipelines, templates, input sets, services, environments, infrastructure definitions, and service overrides. These parameters were optional before but made mandatory now as the APIs require these to work. (CDS-94245)

#### New features and enhancements

> - Harness will enable Overrides (V2) in your accounts and migrate your existing Overrides (V1) to the new V2 experience. Contact [Harness Support](mailto:support@harness.io) for queries or assistance regarding the migration.

Reviewer 2: @roshnisarangadharan can you mention the date as starting April 20th
